packages/zowe-explorer/src/job/utils.ts

```diff
@@ -27,14 +27,16 @@ export const JOB_SORT_OPTS = [
     localize("jobs.sortByDateSubmitted", "$(calendar) Date Submitted"),
     localize("jobs.sortByName", "$(case-sensitive) Job Name"),
     localize("jobs.sortByReturnCode", "$(symbol-numeric) Return Code"),
+    localize("jobs.sortByTimeStamp", "$(clock) Time Stamp"),
     localize("setSortDirection", "$(fold) Sort Direction"),
 ];
 
-export const JOB_SORT_KEYS: Record<JobSortOpts, keyof (IJob & { "exec-submitted": string })> = {
+export const JOB_SORT_KEYS: Record<JobSortOpts, keyof (IJob & { "exec-submitted": string } & { "exec-ended": string })> = {
```

Comment:
Can we combine `exec-submitted` and `exec-ended` into the same type?
This will make it easier to add to the list of options that aren't in `IJob`.

```suggestion
export const JOB_SORT_KEYS: Record<JobSortOpts, keyof (IJob & { "exec-submitted": string; "exec-ended": string })> = {
```

Or, for readability we can extract this object into an interface:

```suggestion
interface IJobExtendedOpts {
    "exec-ended": string;
    "exec-submitted": string;
};
export const JOB_SORT_KEYS: Record<JobSortOpts, keyof (IJob & IJobExtendedOpts)> = {
```

Reply:
Added interface